### DISCUSSION TITLE
Any good example to demonstrate the difference between createMemo vs createComputed and createEffect vs createRenderEffect ?

### DISCUSSION THREAD

**Discussion by cyfung1031** [solidjs/solid#2168] [General]
Any good example to demonstrate the difference between createMemo vs createComputed and createEffect vs createRenderEffect ?

**Comment by AFatNiBBa** [comment]
Check [this](https://playground.solidjs.com/anonymous/a5ebd82e-0a0b-4061-962a-f6893cd88670) playground

**Reply by cyfung1031** [reply]
> Check [this](https://playground.solidjs.com/anonymous/a5ebd82e-0a0b-4061-962a-f6893cd88670) playground

Nice! This is good.

Regarding 
```
(I don't know why but `createComputed()` will be runned before `createMemo()`)
```

[modified](https://playground.solidjs.com/anonymous/d3e00099-06f8-4341-894f-590727327999)

In PART B of the above code, I swapped 6, 7, and 8, then 3 will be executed first.  (seems "createComputed" is affected by "createRenderEffect")

**Reply by wegylexy** [reply]
> In PART B of the above code, I swapped 6, 7, and 8, then 3 will be executed first. (seems "createComputed" is affected by "createRenderEffect")

I tried inserting more `createComputed` at various places. It seems that both `createdComputed` and `createMemo` run mixed together in the same batch as if they are optimized in a hash map and lost their original order. But it was a quick random test, not conclusive.

Hope the author provides a lifecycle diagram, or indicates what may be undefined (if Ryan wishes to leave himself with future flexibility).
